zerver/views/streams.py

```diff
@@ -575,12 +576,13 @@ def add_subscriptions_backend(
         permission_configuration = Stream.stream_permission_group_settings[
             "can_remove_subscribers_group"
         ]
-        can_remove_subscribers_group = access_user_group_for_setting(
-            can_remove_subscribers_group_id,
-            user_profile,
-            setting_name="can_remove_subscribers_group",
-            permission_configuration=permission_configuration,
-        )
+        with transaction.atomic():
```

Comment:
I'm not sure I understand the strategy for this transaction being so short; it seems like the transaction would end before actually saving the new channel to the database?

Reply:
This was because `access_user_group_for_setting` uses `select_for_update` and I wanted to avoid other changes which might break something, so just added the transaction for the specific call. But now the whole function is already inside a transaction, so removed that.